### DISCUSSION TITLE
Suggestion: Adding AppShell

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [shadcn-ui/ui#563] [General]
The Mantine lib has a great component called AppShell [link](https://mantine.dev/core/app-shell/): a sidebar, header, footer, ...etc, I think adding a similar component will be a great addition. since most apps need such a component.

**Comment by kinthaiofficial** [comment]
An AppShell pattern for shadcn/ui would fill a real gap — right now you have to compose the sidebar + header + content area yourself from Primitives, which is repetitive across projects.

**What a good AppShell needs:**
- **Navigation sidebar**: collapsible, responsive, with active state indicators, nested navigation, icons
- **Header/topbar**: breadcrumbs, search, user menu, notification bell
- **Main content area**: proper overflow handling, consistent max-width, padding
- **Mobile drawer**: the sidebar should transform into a bottom sheet or slide-in drawer on mobile

**The design tension:**
shadcn/ui's philosophy is "copy-paste into your project, own it completely." AppShell would be more opinionated than a primitive — it has implicit structure (sidebar on left, header on top, etc.) that doesn't match every app.

**A possible approach:**
Rather than one `AppShell` component, a family of composable shell components:
- `<Shell>`: the outer wrapper (flex/grid container)
- `<Shell.Sidebar>`: the navigation panel
- `<Shell.Header>`: the topbar
- `<Shell.Content>`: the main scrollable area

Each is individually copy-pasted and composable, but there's a reference example showing how they fit together.

**What already exists:**
Shadcn's examples page (shadcn.com/examples) has a "Mail", "Dashboard", and "Music" example that basically implement AppShell patterns. These could be extracted into a formal AppShell component set.

The biggest value-add would be solving the sidebar collapse + mobile breakpoint handling, which has surprising complexity.
